core/app/models/spree/product.rb

```diff
@@ -75,6 +75,20 @@ def find_or_build_master
       master || build_master
     end
 
+    def digital_assets
```

Comment:
Can this be rewritten using AR associations or Arel? Right now it can't be lazy loaded, and it's not cached so it will run the SQL on every invocation. I think it's 3 runs on the admin/images template, but I would be more concerned if the method was used in the frontend.

Reply:
yeah, kinda was expecting that feedback @fastjames ... 😄 - I don't think we have the AR models to do it strictly with AR, but Arel may be do-able. I'll work on it today and see what I can do ...

Reply:
updated with Arel instead of raw SQL @fastjames ...